test/mock/rtps/Endpoint/fastdds/rtps/Endpoint.h

```diff
@@ -34,20 +36,22 @@ class Endpoint
 
     inline RecursiveTimedMutex& getMutex()
     {
-        return mutex_;
+        return mp_mutex;
     }
 
     EndpointAttributes& getAttributes()
     {
-        return attributes_;
+        return m_att;
     }
 
 #if HAVE_SECURITY
     bool supports_rtps_protection_;
 #endif // HAVE_SECURITY
 
-    RecursiveTimedMutex mutex_;
-    EndpointAttributes attributes_;
+    mutable RecursiveTimedMutex mp_mutex;
+    EndpointAttributes m_att;
+    RTPSParticipantImpl* mp_RTPSParticipant;
+
```

Comment:
Attribute name in the mock did not match attribute name in the real class, and some derived/friend class was complaining that it could not find the attribute.